uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/Form/Search/StaffDirectoryTest.php

```diff
@@ -379,6 +379,48 @@ public function testOnlyActiveContactManagersAreReturnedForTheManagerColumn() {
     $this->assertEquals($expectedResults, $results);
   }
 
+  public function testRightContactsAreReturnedWhenIncludeContactIsTrue() {
+    $contact1 = ContactFabricator::fabricate(['first_name' => 'Contact1']);
+    $contact2 = ContactFabricator::fabricate(['first_name' => 'Contact2']);
+    $contact3 = ContactFabricator::fabricate(['first_name' => 'Contact3']);
+
+    $formValues = [];
+    //We need to simulate the values here when a contact is selected via the
+    //checkbox via the UI.
+    $formValues['mark_x_'. $contact1['id']] = '';
+    $formValues['mark_x_'. $contact3['id']] = '';
+
+    $searchDirectory =  new SearchDirectory($formValues);
+    $includeContactId = TRUE;
+    $results = $this->extractColumnValues($searchDirectory->all(0, 0, NULL, $includeContactId));
+
+    //Only contact1 and contact3 are expected
+    $expectedResults = [
+      [
+        'contact_id' => $contact1['id'],
+        'display_name' => $contact1['display_name'],
+        'work_phone' => NULL,
+        'work_email' => NULL,
+        'manager' => NULL,
+        'location' => NULL,
+        'department' => NULL,
+        'job_title' => NULL,
+      ],
+      [
+        'contact_id' => $contact3['id'],
+        'display_name' => $contact3['display_name'],
+        'work_phone' => NULL,
+        'work_email' => NULL,
+        'manager' => NULL,
+        'location' => NULL,
+        'department' => NULL,
+        'job_title' => NULL,
+      ]
+    ];
+
+    $this->assertEquals($expectedResults, $results);
+  }
+
```

Comment:
How about adding a test to make sure the form will return the tasklist with only the expected items?